### PR TITLE
chore(types): narrow string to Locale where possible

### PR DESCRIPTION
**Pull request by C17AN**
## Summary

- implements #11322

### Problem

As the locale key value is widely used in the project, I thought it would be better to manage it more thoroughly.

### Solution

Narrow `locale` property's type from `string` to `Locale`.

### Before

image

### After

image

### CODE REVIEW

**Comment by C17AN** [comment]
I applied the changes that are less affected, but if you think it's better to change them at once, or if you don't think this change is necessary, please feel free to let me know your opinion.

**Review by caugner — APPROVED** [review]
Overall LGTM, but two nits.

File: content/redirect.ts [hunk 1]
```diff
@@ -67,7 +68,7 @@ function validateFromURL(url: string, locale: string, checkPath = true) {
   if (!url.includes("/docs/")) {
     throw new Error(`From-URL must contain '/docs/' was ${url}`);
   }
-  if (!VALID_LOCALES_SET.has(url.split("/")[1])) {
+  if (!VALID_LOCALES_SET.has(url.split("/")[1] as Locale)) {
```
[the remark this answers is not in the document]
Comment (caugner) [reply content/redirect.ts]: Fixed in 42c761f57f.

File: markdown/m2h/cli.ts [hunk 2]
```diff
@@ -44,7 +45,7 @@ program
 
   .option("--locale", "Targets a specific locale", {
     default: "all",
-    validator: Array.from(VALID_LOCALES.values()).concat("all"),
+    validator: Array.from(VALID_LOCALES.values()).concat("all" as Locale),
```
[the remark this answers is not in the document]
Comment (C17AN) [reply markdown/m2h/cli.ts]: I tried not to alter the original form as much as possible, but there was a much better way.
Thank you for finding better way!

**Comment by C17AN** [comment]
Thank you for your review, @caugner!
If there is anything else I need to do, please feel free to let me know, and have a great day!

**Pull request merged**